client/src/components/layout/BuildForm.js

```diff
@@ -21,69 +21,68 @@ const BuildForm = (props) => {
 
     const [buildFormErrors, setBuildFormErrors] = useState({})
 
-    const [redirect, setRedirect] = useState(false)
-
-    const [newBuildID, setBuildID] = useState(0)
+    const [redirect, setRedirect] = useState({
+        redirectState: false,
+        newBuildID: 0
```

Comment:
`null` or `undefined` might be a better default state than an integer for the newBuildId as to indicate that there is currently _no_ build id in state